### PR TITLE
New doc: Drupal 9 early access

### PR DESCRIPTION
**Pull request by EdwardAngert**
## Summary

**[Drupal 9 Early Access](https://pantheon.io/docs/drupal-9)** - Drupal 9 is now available for Early Access testing. New doc outlines how to get started. [PR 5733](https://github.com/pantheon-systems/documentation/pull/5733)

## Effect

The following changes are already committed:

- transcribe and organize MMLC's doc to markdown

## Remaining Work

The following changes still need to be completed:

- [x] Review from @populist and @stevector
- [x] copy review

## Post Launch

**Do not remove** - To be completed by the docs team upon merge:

- [ ] Index /drupal-9 page w/docs search service provider
- [ ] Update Status Report
- [ ] Remove from the [project board](https://github.com/pantheon-systems/documentation/projects/14)

### CODE REVIEW

**Comment by pantheondocs** [comment]
Lighthouse Audit Results:

| Description | Live | pr-5733 | Difference |
| --- | --- | --- | --- |
| Performance Score | `56.00000000000001` | `45` | `-11` |
| Best Practice Score | `93` | `86` | `-7` |
| SEO Score | `96` | `96` | `0` |
| Accessibility Score | `98` | `96` | `-2` |

Lighthouse Audit Reports:

- [`pr-5733` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/21162/artifacts/0//lighthouse_data/lighthouse-audit-dev.html)
- [`Pantheon Docs` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/21162/artifacts/0//lighthouse_data/lighthouse-audit-reference.html)

**Comment by populist** [comment]
I added some additional commits to the doc. Loving getting this rolling!

We should decide on if we want to move the two D9 repos off @stevector's GitHub to Pantheon's and then we can update those links.

Otherwise looks great to me. Looking forward to updating this once we get out of Early Access and into general release!

File: source/content/drupal-9.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+Want to regularly develop with continuous integration (CI) and automated tests? Leverage Pantheon’s [Build Tools](/guides/build-tools) to get up and running quickly with a CI workflow (CircleCI, BitBucket, GitLab), to build, test, and deploy D9.
+
+1. Install the [Terminus Build Tools plugin](https://github.com/pantheon-systems/terminus-build-tools-plugin).
+   - This will require generating a Pantheon machine token, a CircleCI API token, and a GitHub API credential.
+1. Enable support for custom CircleCI Orbs in your Organizational settings. In CircleCI, click **Settings**, then **Security**, and click **Yes** to allow uncertified orbs:
+   ![Allow Uncertified Orbs in CircleCI](../images/circleci/circleci-allow-uncertified-orbs.png)
+1. Use Terminus to create a new D9 site, complete with a customizable CI workflow. Remember to replace `<NEW-SITE-NAME>` in this example with your new site name:
+
+  ```bash{promptUser: user}
+  terminus build:project:create stevector/drupal-9-with-pantheon-orb <NEW-SITE-NAME> --stability=dev
```
Comment (stevector) [inline source/content/drupal-9.md:42]: We have a new repo under `pantheon-systems` now. Please use

```
terminus build:project:create "pantheon-systems/drupal-9-with-circleci-orb:dev-master" <NEW-SITE-MACHINE-NAME> --stability=dev --team='optional-pantheon-organization-name' 
```

Comment (carolyn-shannon) [reply source/content/drupal-9.md:42]: Thanks @stevector. Is your review complete & approved? It sounds like @populist has approved, so once we have the thumbs-up from you we can do copy review & publish.

**Comment by stevector** [comment]
@carolyn-shannon I'm reviewing top to bottom now

File: source/content/drupal-9.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+  ```
+
+1. Clone your D8 [site’s codebase to your computer](https://pantheon.io/docs/local-development#get-the-code). You can create a new D8 site or use an existing D8 site:
```
Comment (stevector) [inline source/content/drupal-9.md:45]: Lines 45 through 81 seem unrelated to the current heading. Looking back at the source google doc, these seem like a copy/paste error in the original doc. These seem like an alternate version of `Test Upgrade an Existing Pantheon D8 Site to D9`. I think these can be deleted.

https://docs.google.com/document/d/1ENHYy5hLOtZhoKfIITx2RvkrvN7v9oYk6TrFDsmfONk/edit#heading=h.ytrq2fmbkmxq

File: source/content/drupal-9.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+1. Create and switch to a new testing branch in the D8 codebase:
+
+  ```bash{promptUser: user}
+  git checkout -b d9-beta
+  ```
+
+1. Pull the `drupal9-beta3-early-access-not-for-production-use` branch into your codebase:
+
+  ```bash{promptUser: user}
+  git pull https://github.com/pantheon-systems/drops-8.git drupal9-beta3-early-access-not-for-production-use
```
Comment (stevector) [inline source/content/drupal-9.md:134]: I think this tag was made with a commit on top of 8.8.5. So for this part to work cleanly I had to be on a git history with Drupal 8.8.5. When I tried this when on a site on 8.8.6 I got a huge merge conflict. I thought that adding `-X theirs` would work but it did not. @greg-1-anderson any tips?

Comment (greg-1-anderson) [reply source/content/drupal-9.md:134]: Yes, this is how it was expected to work (or not work, as the case may be).

We could rewrite the 9.0.0-beta3 tag on top of the 8.8.6 release, if desired.

File: source/content/drupal-9.md [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+reviewed: "2020-05-21"
+---
+
+Drupal 9 is, according to [drupal.org](https://www.drupal.org/docs/understanding-drupal/drupal-9-resources), "a cleaned-up version of Drupal 8... \[with] deprecated code removed and third-party dependencies updated."
+
+D9 updates Drupal’s underlying dependencies like [Symfony 4.4](https://symfony.com/releases/4.4) and [Twig 2](https://twig.symfony.com/doc/2.x/index.html), removes several deprecated API functions in favor of better options, and allows everyone running Drupal 8.8+ an easy upgrade path to D9 & beyond.
+
+<Alert title="Warning" type="danger">
+
+Currently, Drupal 9 is only available on Pantheon as a Early Access feature release. This means the feature is currently in active development and is intended for testing and evaluation purposes only. **Sites may be deleted without notice**.
```
[the remark this answers is not in the document]
Comment (greg-1-anderson) [reply source/content/drupal-9.md:17]: Regarding the merge conflict problem described below, that's going to happen in both directions. That is, if someone took their site from 8.8.5 or 8.8.6 (whatever we base the 9.0.0-beta3 tag on), and then later want to dashboard-update or even just git-merge-update their site from the pre-release to the stable release, they are going to get the kind of merge conflicts that -X theirs is not going to help.

We should have some sort of warning about this. We might decide to provide a conversion tool to help with this transition, but I think it would be better to impart the understanding that Drupal 9 environments are throwaway. That is, if you make a new Drupal 9 site, we may delete it. If you make a multidev environment from your production site and switch it to Drupal 9, we won't delete the whole site, but neither should we guarantee that environment is going to keep running correctly after the 9.0.0 stable release.

File: source/content/drupal-9.md (hunk 1 above)
Comment (greg-1-anderson) [reply source/content/drupal-9.md:42]: Is pantheon-systems/drupal-9-with-circleci-orb going to be maintained? The README currently says it is experimental. Perhaps you should remove the composer.lock file from the repository. Then the only reason this repo would need to be updated would be if something were to break.

File: source/content/drupal-9.md [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+1. Clone your D8 [site’s codebase to your computer](https://pantheon.io/docs/local-development#get-the-code). You can create a new D8 site or use an existing D8 site:
+
+  ```bash{promptUser: user}
+  git clone <url-for-site-repo>
+  ```
+
+1. Use Terminus to create a [Multidev](/multidev) environment called `update-9.0.0-beta3` on your D8 site for testing:
+
+  ```bash{promptUser: user}
+  terminus multidev:create <site>.<env> update-9.0.0-beta3
```
Comment (greg-1-anderson) [inline source/content/drupal-9.md:54]: Multidev names can only be 11 characters long.

File: source/content/drupal-9.md [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+
+  ```bash{promptUser: user}
+  git pull https://github.com/pantheon-systems/drops-8.git drupal9-beta3-early-access-not-for-production-use
+  ```
+
+1. Modify the `pantheon.yml` file to specify PHP 7.4 and Drush 8:
+
+  ```yaml:title=pantheon.yml
+  php_version: 7.4
+  drush_version: 8
```
Comment (greg-1-anderson) [inline source/content/drupal-9.md:141]: n.b. we had to back out Drush 8.3.3 because it caused problems with the update to the Drupal 8.8.6 security release. Drush 8.3.2 does not work with Drupal 9.

Drush 8 is still the best advice, but note that it does not work today.

File: source/content/drupal-9.md (hunk 5 above)
Comment (stevector) [reply source/content/drupal-9.md:54]: Yes, and this is within a section that should be deleted entirely. See https://github.com/pantheon-systems/documentation/pull/5733#discussion_r428406329

File: source/content/drupal-9.md (hunk 3 above)
Comment (stevector) [reply source/content/drupal-9.md:134]: Yes, let's make another one. How about the name `drupal9-beta3--from-8.8.6--early-access-not-for-production-use`

File: source/content/drupal-9.md (hunk 1 above)
Comment (stevector) [reply source/content/drupal-9.md:42]: @greg-1-anderson, I expect it to maintain it as an example of a minimal CI+D9 set up. I think there might be a need a for a fuller `example-drops-9-composer` at some point. Here's a PR with an updated Readme: https://github.com/pantheon-systems/drupal-9-with-circleci-orb/pull/2

File: source/content/drupal-9.md [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+
+Join our early access program to test out both D9 and Pantheon native Composer support.
+
+1. [Fill out this form](https://docs.google.com/forms/d/1lahWKMT2VHXfr9hg15VIQY2Kn6z_j77o7Te6hZqsNgw) to get access to our “Drupal 9 Early Access” group. Once you are added, you will access to a new "Drupal 9 Early Access" site creation option.
+
+1. Spin up a new "Drupal 9 Early Access" site with `ic-demo-2020-` as a site name prefix (i.e., `ic-demo-2020-beta3-d9-test`). This will give you [D9 starter codebase](https://github.com/stevector/drupal-9-project) and will automatically trigger a Composer install and `git commit` on site creation.
+
+## Create a D9 Site with Continuous Integration via Build Tools
+
+Want to regularly develop with continuous integration (CI) and automated tests? Leverage Pantheon’s [Build Tools](/guides/build-tools) to get up and running quickly with a CI workflow (CircleCI, BitBucket, GitLab), to build, test, and deploy D9.
```
[the remark this answers is not in the document]
Comment (EdwardAngert) [reply source/content/drupal-9.md:33]: I took this to be an example of CIs, but I see what you're saying - incoming PR with that removed, and "These steps use GitHub and CircleCI" to clarify

File: source/content/drupal-9.md (hunk 6 above)
Comment (EdwardAngert) [reply source/content/drupal-9.md:112]: thanks, @greg-1-anderson 
1. should I modify the branch name to drop the `.8.6` or is that something that needs to be changed elsewhere for this to work?
1. should we mention that php 7.3 is also an option?

File: source/content/drupal-9.md [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+  git checkout -b d9-beta
+  ```
+
+1. Pull the `drupal9-beta3--from-8.8.6--early-access-not-for-production-use` branch into your codebase:
+
+  ```bash{promptUser: user}
+  git pull https://github.com/pantheon-systems/drops-8.git drupal9-beta3--from-8.8.6--early-access-not-for-production-use
+  ```
+
+1. Modify the `pantheon.yml` file to specify PHP 7.4 and Drush 8:
```
Comment (ari-gold) [inline source/content/drupal-9.md:108]: PHP 7.3 is also ok

File: source/content/drupal-9.md [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+categories: [get-started]
+tags: [site]
+earlyaccess: true
+contributors: [populist, edwardangert]
+reviewed: "2020-05-21"
+---
+
+Drupal 9 is, according to [drupal.org](https://www.drupal.org/docs/understanding-drupal/drupal-9-resources), "a cleaned-up version of Drupal 8... \[with] deprecated code removed and third-party dependencies updated."
+
+D9 updates Drupal’s underlying dependencies like [Symfony 4.4](https://symfony.com/releases/4.4) and [Twig 2](https://twig.symfony.com/doc/2.x/index.html), removes several deprecated API functions in favor of better options, and allows everyone running Drupal 8.8+ an easy upgrade path to D9 & beyond.
```
[the remark this answers is not in the document]
Comment (EdwardAngert) [reply source/content/drupal-9.md]: adjusted these - may have to update headings in the blog post links

File: source/content/drupal-9.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (EdwardAngert) [reply source/content/drupal-9.md]: I updated this copy to

```
**Do not use this, or any, Early Access release for your live site.**

Currently, Drupal 9 is only available on Pantheon as an Early Access feature release. This means the feature is currently in active development and is intended for testing and evaluation purposes only.

Pantheon engineers are rolling out changes often, and easy updates from Drupal 9 Early Access are not guaranteed.

**Sites may be deleted without notice**.
```

File: source/content/drupal-9.md [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+
+Choose one of the following methods to get started with a new D9 site, to test your existing site's [compatibility with an upgrade](#test-an-existing-drupal-site-for-drupal-9-upgrade-compatibility), or to [test run your Pantheon Drupal 8 site on D9](#test-upgrade-an-existing-pantheon-drupal-8-site-to-drupal-9).
+
+## Create a New D9 Site with Integrated Composer
+
+Join our early access program to test out both D9 and Pantheon native Composer support.
+
+1. [Fill out this form](https://docs.google.com/forms/d/1lahWKMT2VHXfr9hg15VIQY2Kn6z_j77o7Te6hZqsNgw) to get access to our “Drupal 9 Early Access” group. Once you are added, you will access to a new "Drupal 9 Early Access" site creation option.
+
+1. Spin up a new "Drupal 9 Early Access" site with `ic-demo-2020-` as a site name prefix (i.e., `ic-demo-2020-beta3-d9-test`). This will give you [D9 starter codebase](https://github.com/stevector/drupal-9-project) and will automatically trigger a Composer install and `git commit` on site creation.
```
[the remark this answers is not in the document]
Comment (EdwardAngert) [reply source/content/drupal-9.md]: the warning is up top

File: source/content/drupal-9.md [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+
+Choose one of the following methods to get started with a new Drupal 9 site to [create a new Drupal 9 site with integrated Composer](#create-a-new-drupal-9-site-with-integrated-composer) or to test your existing site's [compatibility with an upgrade](#test-an-existing-drupal-site-for-drupal-9-upgrade-compatibility).
+
+## Create a New Drupal 9 Site with Integrated Composer
+
+Join our Early Access program to test out both Drupal 9 and Pantheon-native Composer support.
+
+1. [Fill out this form](https://docs.google.com/forms/d/1lahWKMT2VHXfr9hg15VIQY2Kn6z_j77o7Te6hZqsNgw) to get access to our “Drupal 9 Early Access” group. Once you are added, you will access to a new "Drupal 9 Early Access" site creation option.
+
+1. Spin up a new "Drupal 9 Early Access" site with `ic-demo-2020-` as a site name prefix (i.e., `ic-demo-2020-beta3-d9-test`). This will give you [Drupal 9 starter codebase](https://github.com/stevector/drupal-9-project) and will automatically trigger a Composer install and `git commit` on site creation.
```
Comment (EdwardAngert) [inline source/content/drupal-9.md:35]: @stevector do you still want this and L91 pointing to github/stevevector?

Comment (stevector) [reply source/content/drupal-9.md:35]: Yeah, I'm fine with leaving it since this is the Early Access repo and won't be used  long-term. We can move the repo later if we need to and GitHub will redirect.

File: source/content/drupal-9.md [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+
+Currently, Drupal 9 is only available on Pantheon as an Early Access feature release. This means the feature is currently in active development and is intended for testing and evaluation purposes only.
+
+Pantheon engineers are rolling out changes often, and easy updates from Drupal 9 Early Access are not guaranteed.
+
+**Sites may be deleted without notice**.
+
+</Alert>
+
+Choose one of the following methods to get started with a new Drupal 9 site to [create a new Drupal 9 site with integrated Composer](#create-a-new-drupal-9-site-with-integrated-composer) or to test your existing site's [compatibility with an upgrade](#test-an-existing-drupal-site-for-drupal-9-upgrade-compatibility).
```
Comment (greg-1-anderson) [inline source/content/drupal-9.md:27]: It might be helpful to give people some guidance or advice on which option to select.

- Using the "Drupal 9 Early Access" form is best for those who want to try out the new Integrated Composer feature. This is also probably the best option right now for folks who just want to try Drupal 9 with the least overhead.
- Use `build:project:create` if already using the Build Tools workflow. I'd also say this option was best for those interested in automated testing, but I don't think any automated tests are working yet.
- Even though we don't have a section for the drops-8 branch yet, we might advise those not using Composer just wait & use the branch once it becomes available.

This advice could either come before the options, after the options, or be divided up so that each section has the blerb about when it is most applicable to use it.

Comment (stevector) [reply source/content/drupal-9.md:27]: +1, I recommend rewrite this sentence as something like 

> You can test Drupal 9 now with a fresh installation using either our Integrated Composer Early Access program or Build Tools. We recommend Integrate Composer if you simply want to test the browser experience of Drupal 9 and get access to forthcoming Pantheon support for Composer in the Dashboard. Use Build Tools if you want to create a continuous integration workflow using GitHub. If you are more concerned with upgrading an existing site, you can check compatibility with an upgrade.

File: source/content/drupal-9.md [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+Drupal 9:
+
+```none
+use \Drupal\node\Entity\Node;
+$node = Node::load(1);
+```
+
+Since most of these changes are relatively minor, there are a number of [deprecation checking and correction tools](https://www.drupal.org/docs/9/how-to-prepare-your-drupal-7-or-8-site-for-drupal-9/deprecation-checking-and-correction-tools) available.
+
+## FAQ
```
Comment (ari-gold) [inline source/content/drupal-9.md:80]: Does Pantheon have plans to upgrade MariaDB?

Yes, Pantheon has plans to upgrade MariaDB platform-wide before September 30, 2020.

File: source/content/drupal-9.md [hunk 14]
```diff
@@ -0,0 +1,91 @@
+---
+title: Drupal 9 Early Access
+description: Test the latest Drupal on the Pantheon Website Management Platform.
```
Comment (ari-gold) [inline source/content/drupal-9.md:3]: ..on the Pantheon WebOps Platform

File: source/content/drupal-9.md [hunk 15]
```diff
@@ -0,0 +1,91 @@
...
+reviewed: "2020-05-21"
+---
+
+Drupal 9 is, according to [drupal.org](https://www.drupal.org/docs/understanding-drupal/drupal-9-resources), "a cleaned-up version of Drupal 8... \[with] deprecated code removed and third-party dependencies updated."
+
+Drupal 9 updates Drupal’s underlying dependencies like [Symfony 4.4](https://symfony.com/releases/4.4) and [Twig 2](https://twig.symfony.com/doc/2.x/index.html), removes several deprecated API functions in favor of better options, and allows everyone running Drupal 8.8+ an easy upgrade path to Drupal 9 & beyond.
+
+<Alert title="Warning" type="danger">
+
+**Do not use this, or any, Early Access release for your live site.**
```
Comment (ari-gold) [inline source/content/drupal-9.md:17]: remove ", or any,"

File: source/content/drupal-9.md [hunk 16]
```diff
@@ -0,0 +1,91 @@
...
+
+Currently, Drupal 9 is only available on Pantheon as an Early Access feature release. This means the feature is currently in active development and is intended for testing and evaluation purposes only.
+
+Pantheon engineers are rolling out changes often, and easy updates from Drupal 9 Early Access are not guaranteed.
```
Comment (ari-gold) [inline source/content/drupal-9.md:21]: suggest deleting this sentence

File: source/content/drupal-9.md [hunk 17]
```diff
@@ -0,0 +1,91 @@
...
+earlyaccess: true
+contributors: [populist, edwardangert]
+reviewed: "2020-05-21"
+---
+
+Drupal 9 is, according to [drupal.org](https://www.drupal.org/docs/understanding-drupal/drupal-9-resources), "a cleaned-up version of Drupal 8... \[with] deprecated code removed and third-party dependencies updated."
+
+Drupal 9 updates Drupal’s underlying dependencies like [Symfony 4.4](https://symfony.com/releases/4.4) and [Twig 2](https://twig.symfony.com/doc/2.x/index.html), removes several deprecated API functions in favor of better options, and allows everyone running Drupal 8.8+ an easy upgrade path to Drupal 9 & beyond.
+
+<Alert title="Warning" type="danger">
```
Comment (ari-gold) [inline source/content/drupal-9.md:15]: Does the heading need to say "warning" ? Can we move the below in bold into the heading?

"Early Access Only - Not for Production Use"

File: source/content/drupal-9.md [hunk 18]
```diff
@@ -0,0 +1,91 @@
...
+
+**Do not use this, or any, Early Access release for your live site.**
+
+Currently, Drupal 9 is only available on Pantheon as an Early Access feature release. This means the feature is currently in active development and is intended for testing and evaluation purposes only.
+
+Pantheon engineers are rolling out changes often, and easy updates from Drupal 9 Early Access are not guaranteed.
+
+**Sites may be deleted without notice**.
```
Comment (ari-gold) [inline source/content/drupal-9.md:23]: Let's move this into the Integrated Composer section.

File: source/content/drupal-9.md (hunk 12 above)
Comment (ari-gold) [reply source/content/drupal-9.md:27]: How about copying the table from the blog post into here?

File: source/content/drupal-9.md [hunk 19]
```diff
@@ -0,0 +1,91 @@
...
+
+</Alert>
+
+Choose one of the following methods to get started with a new Drupal 9 site to [create a new Drupal 9 site with integrated Composer](#create-a-new-drupal-9-site-with-integrated-composer) or to test your existing site's [compatibility with an upgrade](#test-an-existing-drupal-site-for-drupal-9-upgrade-compatibility).
+
+## Create a New Drupal 9 Site with Integrated Composer
+
+Join our Early Access program to test out both Drupal 9 and Pantheon-native Composer support.
```
Comment (ari-gold) [inline source/content/drupal-9.md:31]: I'd like to avoid "native" if possible -- can we just say integrated composer?

File: source/content/drupal-9.md (hunk 12 above)
Comment (EdwardAngert) [reply source/content/drupal-9.md]: pushed a commit with this in bullet points - testing with the table while it builds a preview

File: source/content/drupal-9.md (hunk 16 above)
Comment (EdwardAngert) [reply source/content/drupal-9.md:21]: I'm hesitant to remove this as it's a soft warning that users might not be able to update once they're on D9. Given the other warnings up top, I don't think we _have_ to keep it, but I like it as a reminder that folks shouldn't expect updates to definitely work.

leaving for copy editor

File: source/content/drupal-9.md (hunk 18 above)
Comment (EdwardAngert) [reply source/content/drupal-9.md]: left this above the sections because it's for both, leaving this as not-resolved for now

File: source/content/drupal-9.md (hunk 12 above)
Comment (EdwardAngert) [reply source/content/drupal-9.md]: table added as well

File: source/content/drupal-9.md (hunk 6 above)
Comment (EdwardAngert) [reply source/content/drupal-9.md]: resolving - doesn't apply to this pr
